### DISCUSSION TITLE
Is each websocket connection only expected to have one `connection` at a time

### DISCUSSION THREAD

**Discussion by edwardfeng-db** [enisdenjo/graphql-ws#158] [Q&A] [answered]
Hey there,

I'm a bit confused on the concept of `websocket connection` and `connection` using `graphql-ws`. Seems like you use `connection_init` and `connection_ack` to start a `connection` within an established websocket connection. However, I'm wondering is it expected for a websocket connection to only have one `connection` at a time? Or you can have multiple connections? Thanks!

**Comment by enisdenjo** [comment, marked as answer]
The WS connection and `graphql-ws` connection are synonymous, the only difference is that a `graphql-ws` connection is an **acknowledged** connection (through the `conneciton_ack` message) within a WS socket. Check [the `Nomenclature` section in the Protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#nomenclature).

You cannot have 2 connections within a single socket the same way you cannot have 2 sockets within one socket.

**Reply by edwardfeng-db** [reply]
Got it. Thanks @enisdenjo!
